cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -1016,17 +1016,44 @@ set(FC_DECLARE_COMMON_OPTIONS)
 if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.28)
   list(APPEND FC_DECLARE_COMMON_OPTIONS EXCLUDE_FROM_ALL TRUE)
 endif()
+if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.25)
+  list(APPEND FC_DECLARE_COMMON_OPTIONS SYSTEM)
+endif()
 
 macro(prepare_fetchcontent)
-  set(BUILD_SHARED_LIBS OFF)
-  set(BUILD_STATIC_LIBS ON)
-  set(CMAKE_COMPILE_WARNING_AS_ERROR FALSE)
-  set(CMAKE_EXPORT_NO_PACKAGE_REGISTRY TRUE)
-  set(CMAKE_MACOSX_RPATH ${ARROW_INSTALL_NAME_RPATH})
+  set(BUILD_SHARED_LIBS
```

Comment:
Can you add a comment explaining why these have to be CACHE variables? AFAIU, CACHE variables have a lot of side effects, so it's better to avoid them?

Reply:
OK. I've added a comment for it: https://github.com/apache/arrow/pull/45306/commits/874363dbffb748645c241990a828d7b6fabe28a8

It's related to `CMP0077`: https://cmake.org/cmake/help/latest/policy/CMP0077.html

> [option()](https://cmake.org/cmake/help/latest/command/option.html#command:option) honors normal variables.